doc/refman/RefMan-ssr.tex

```diff
@@ -0,0 +1,4687 @@
+\begin{lstlisting}
+y : nat
+==================
+true && (y < 2) = true
+\end{lstlisting}
+If the cleared names are reused in the same intro pattern, a renaming
+is performed behind the scenes.
+
+Facts mentioned in a clear switch must be valid
+names in the proof context (excluding the section context).
+
+The rules for interpreting branching and destructing $\ssrN{i-pattern}$
+are motivated by the fact that it would be pointless to have a branching
+pattern if ${\tac}$ is a \ssrC{move}, and in most of the remaining cases
+${\tac}$ is \ssrC{case} or \ssrC{elim}, which implies destruction.
+The rules above imply that
+\begin{lstlisting}
+  move=> [a b].
+  case=> [a b].
+  case=> a b.
+\end{lstlisting}
+are all equivalent, so which one to use is a matter of style;
+\ssrC{move} should be used for casual decomposition,
+such as splitting a pair, and \ssrC{case} should be used for actual decompositions,
+in particular for type families (see~\ref{ssec:typefam})
+and proof by contradiction.
+
+The trivial branching $\ssrN{i-pattern}$ can be used to force the branching
+interpretation, e.g.,
+\begin{lstlisting}
+  case=> [] [a b] c.
+  move=> [[a b] c].
+  case;  case=> a b c.
+\end{lstlisting}
+are all equivalent.
+
+\subsection{Generation of equations}\label{ssec:equations}
+
+The generation of named equations option stores the definition of a
+new constant as an equation. The tactic:
+\begin{lstlisting}
+  move En: (size l) => n.
+\end{lstlisting}
+where \ssrC{l} is a list, replaces \ssrC{size l} by \ssrC{n} in the goal and
+adds the fact \ssrC{En : size l = n} to the context.
+ This is quite different from:
+\begin{lstlisting}
+  pose n := (size l).
+\end{lstlisting}
+which generates a definition \ssrC{n := (size l)}. It is not possible to
+generalize or
+rewrite such a definition; on the other hand, it is automatically
+expanded during
+computation, whereas expanding the equation \ssrC{En} requires explicit
+rewriting.
+
+The use of this equation name generation option with a \ssrC{case} or an
+\ssrC{elim} tactic changes the status of the first \iitem{}, in order to
+deal with the possible parameters of the constants introduced.
+
+On the
+goal \ssrC{a <> b} where \ssrC{a, b} are natural numbers, the tactic:
+\begin{lstlisting}
+  case E : a => [|n].
+\end{lstlisting}
+generates two subgoals. The equation \ssrC{E : a = 0}  (resp. \ssrC{E : a =
+  S n}, and the constant \ssrC{n : nat}) has been added to
+the context of the goal \ssrC{0 <> b} (resp. \ssrC{S n  <> b}).
+
+If the user does not provide a branching \iitem{} as first \iitem{},
+or if the \iitem{} does not provide enough names for the arguments of
+a constructor,
+then the constants generated are introduced under fresh \ssr{} names.
+For instance, on the goal \ssrC{a <> b}, the tactic:
+\begin{lstlisting}
+  case E : a => H.
+\end{lstlisting}
+also generates two subgoals, both requiring a proof of \ssrC{False}.
+ The hypot
```

Comment:
I'd say no need to mention them since SSR provides `by` and `done` which serve the same purpose.